contraint.go

```diff
@@ -134,6 +134,94 @@ func (c CustomStringConstraint) ValidateString(value *string, scope Scope) error
 		CreateViolation()
 }
 
+// ConditionalConstraint is used for conditional validation.
+// Use the When function to initiate a conditional check.
+// If the condition is true, then the constraints passed through the Then function will be applied.
+// Otherwise, the constraints passed through the Else function will apply.
+type ConditionalConstraint struct {
+	condition       bool
+	thenConstraints []Constraint
+	elseConstraints []Constraint
+}
+
+// When function using to initiate a conditional check.
+func When(condition bool) ConditionalConstraint {
+	return ConditionalConstraint{
+		condition: condition,
+	}
+}
+
+// Then function applied the constraints if the condition is true.
+//
+// Example
+//  v := "foo"
+//	err := validator.ValidateString(
+//		&value,
+//		validation.When(true).
+//		Then(
+//			it.Matches(regexp.MustCompile(`^\\w$`)),
+//		),
+//	)
+func (c ConditionalConstraint) Then(constraints ...Constraint) ConditionalConstraint {
+	c.thenConstraints = constraints
+	return c
+}
+
+// Else function applied the constraints if the condition is false.
```

Comment:
Else function is used to set a sequence of constraints to be applied if a condition is false.